source/Nevermore/Advanced/Queryable/AsyncEnumerableAdapter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Nevermore.Advanced.Queryable
+{
+    internal static class AsyncEnumerableAdapter
+    {
+        public static async Task<IEnumerable> ConvertToEnumerable(object asyncEnumerable, Type sequenceType, CancellationToken cancellationToken)
+        {
+            var asyncEnumerableType = typeof(IAsyncEnumerable<>).MakeGenericType(sequenceType);
```

Comment:
A tip for reducing the amount of generic hoop-jumping:

Create an actual generic method which does what you want, then instead of doing lots of `MakeGenericType`, do a `MakeGenericMethod` and call it via reflection.

This lets you get rid of CreateList too

e.g.

```csharp
static async Task<IEnumerable> ConvertToEnumerableHelper<T>(IAsyncEnumerable<T> enumerable, CancellationToken ct) {
    var result = new List<T>();
    await foreach(var item in enumerable.WithCancellation(ct)) {
       result.add(item);
    }
    return result;
}

public static Task<IEnumerable> ConvertToEnumerable(object asyncEnumerable, Type sequenceType, CancellationToken cancellationToken) {
   var method = typeof(AsyncEnumerableAdapter).GetMethod("ConvertToEnumerableHelper").MakeGenericMethod(sequenceType);
   
   return (Task<IEnumerable>)method.Invoke(asyncEnumerable);
}
```

Reply:
Good call. I'll tidy it up